packages/core/src/Modules/Cashier/Containers/cashier.jsx

```diff
@@ -18,39 +18,61 @@ import 'Sass/app/modules/cashier.scss';
 
 const el_landscape_blocker = document.getElementById('landscape_blocker');
 
-class Cashier extends React.Component {
-    state = { device_height: window.innerHeight };
-
-    async componentDidMount() {
-        this.props.toggleCashier();
+const Cashier = ({
+    history,
+    is_account_transfer_visible,
+    is_eu,
+    is_onramp_tab_visible,
+    is_p2p_enabled,
+    is_payment_agent_transfer_visible,
+    is_payment_agent_visible,
+    is_virtual,
+    is_visible,
+    location,
+    loggedin_currency,
+    onMount,
+    p2p_notification_count,
+    routes: routes_config,
+    routeBackInApp,
+    setAccountSwitchListener,
+    setTabIndex,
+    tab_index,
+    toggleCashier,
+}) => {
+    const device_height = window.innerHeight;
+
+    React.useEffect(() => {
+        toggleCashier();
         // we still need to populate the tabs shown on cashier
-        await WS.wait('authorize');
+        (async () => {
+            await WS.wait('authorize');
+        })();
```

Comment:
this will just call the function, it won't wait until the function is executed

We should change it to something like this

```
(async () => {
           await WS.wait('authorize');
           onMount();
           setAccountSwitchListener();
          ...
 })();
```